corehq/util/model_log.py

```diff
@@ -1,27 +1,25 @@
 from enum import Enum
 
-from django.contrib.admin.models import ADDITION, CHANGE, DELETION, LogEntry
-from django.contrib.admin.options import get_content_type_for_model
-from django.utils.encoding import force_text
+from corehq.apps.users.models_sql import HQLogEntry, UpdateDetails
 
 
 class ModelAction(Enum):
-    CREATE = ADDITION
-    UPDATE = CHANGE
-    DELETE = DELETION
+    CREATE = HQLogEntry.CREATE
+    UPDATE = HQLogEntry.UPDATE
+    DELETE = HQLogEntry.DELETE
 
 
-def log_model_change(user, model_object, message=None, fields_changed=None, action=ModelAction.UPDATE):
+def log_model_change(domain, user, model_object, message=None, fields_changed=None, action=ModelAction.UPDATE):
     """
-    :param user: User making the change (couch user or django user)
-    :param model_object: The object being changed (must be a Django model)
+    :param domain: domain where the update was initiated
+    :param user: User making the change (couch user)
+    :param model_object: The user being changed (couch user)
     :param message: Message text
     :param fields_changed: List of model field names that have
     :param action: Action on the model
     """
-    from corehq.apps.users.models import CouchUser
-    if isinstance(user, CouchUser):
-        user = user.get_django_user()
+    if not domain:
+        raise ValueError("Please pass domain")
```

Comment:
why the special treatment for `domain`? None of the other fields are validated.

Reply:
This is to catch cases where we don't pass in the domain internally. We should be allowing very specific cases like system updates, or invitation-less registrations to allow domain less log entries.